tests/code_fellows/test_linked_list.py

```diff
@@ -1,21 +1,45 @@
 # https://codefellows.github.io/sea-python-401d6/assignments/linked_list.html
 
+import pytest
+
 from code_fellows.linked_list import LinkedList
 
 
 def test_linked_list():
-    assert LinkedList(["head"]).head.value == "head"
-    assert (
-        LinkedList([3, "three"]).head.value == "head"
-        and LinkedList([3, "three"]).head.next.value == 3
-    )
+    assert LinkedList([3, "three"]).head.value == "three"
+    assert LinkedList([3, "three"]).head.next.value == 3
 
 
 def test_Linked_List_push():
     ll = LinkedList([3, "three"])
     assert ll.head.value == "three"
     assert ll.head.next.value == 3
-    ll.push("new_head")
+    ll.push(["new_head"])
     assert ll.head.value == "new_head"
     assert ll.head.next.value == "three"
     assert ll.head.next.next.value == 3
+
+
+def test_Linked_List_size():
+    ll = LinkedList([3, "three"])
+    assert ll.size() == 2
+    ll = LinkedList(["new_head", 3, "three"])
+    assert ll.size() == 3
+
+
+def test_Linked_List_search():
+    ll = LinkedList([3, "three", "nine"])
+    assert ll.search(3).value == 3
+    expected = ll.head.next.next
+    assert ll.search(3) == expected
+
+
+def test_linked_list_pop():
```

Comment:
That's tricky! I kind of was doing something similar by adding asserts for size() inside of the pop() and push() tests BUT then I remembered I need to isolate everything as much as possible to ensure my tests are actually testing the individual function. Because really, what is the difference between adding an assert in pop() and/or push() to test for size() vs adding an assert in size() to test for pop() and/or push()?

Or should we view classes differently in that regard since all of the functions are part of that one class? Which I guess makes sense. So, in that sense, would it be better suited to test the class as a whole and have a test that just asserts a flow of the functions together? Am I overthinking it?

Reply:
Over thinking it.  `size` should be tested more or less by itself, but anything that could effect the abstract concept of size should include a test for size in some fashion.